### PR TITLE
Support exporting CircleCI pipelines

### PR DESCRIPTION
**Pull request by ZhuGongpu**

### CODE REVIEW

**Comment by ZhuGongpu** [comment]
@Liujingfang1 @monopole @frankfarzan

File: internal/cmdexport/cmdexport_test.go [hunk 1]
```diff
@@ -276,6 +276,70 @@ pipeline {
...
+        commands:
+            run-functions:
+                steps:
+                  - run: kpt fn run resources --fn-path config/functions.yaml
+        jobs:
+            run-functions:
+                executor: kpt-container
+                steps:
+                  - setup_remote_docker
+                  - run-functions
```
Comment (Liujingfang1) [inline internal/cmdexport/cmdexport_test.go:338]: Is the step "run-functions" from the commands "run-functions"? Can you change the commands "run-functions" to "kpt-function-run"?

File: internal/cmdexport/orchestrators/circleci.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+// CircleCIOrb represents a reusable orb object that is a collection of executors, commands, and jobs.
+type CircleCIOrb struct {
+	Executors map[string]*CircleCIExecutor `yaml:",omitempty"`
+	Commands  map[string]*CircleCICommand  `yaml:",omitempty"`
+	Jobs      map[string]*CircleCIJob      `yaml:",omitempty"`
+}
+
+type CircleCIOrbConfig struct {
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/circleci.go:71]: Please add a comment for `CircleCIOrbConfig`.

File: internal/cmdexport/orchestrators/circleci_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		expected: `
+executors:
+    kpt-container:
+        docker:
+          - image: gongpu/kpt:latest
+commands:
+    run-functions:
+        steps:
+          - run: kpt fn run resources --fn-path config/gate-keeper.yaml --fn-path
+                config/label-namespace.yaml
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/circleci_test.go:86]:
```run: kpt fn run resources --fn-path config/gate-keeper.yaml --fn-path
                config/label-namespace.yaml
 ```
I'm curious where you convert it to multiple lines.

Comment (ZhuGongpu) [reply internal/cmdexport/orchestrators/circleci_test.go:86]: `kyaml` breaks the line itself.

Comment (Liujingfang1) [reply internal/cmdexport/orchestrators/circleci_test.go:86]: Ah, good to know.

**Pull request merged**